Automattic/wp-calypso
# Calypso Masterbar: add mini shopping cart

#### Changes proposed in this Pull Request

This adds a persistent shopping cart always available in calypso. It adds a button to the masterbar that allows viewing the cart, deleting items from the cart, and visiting checkout. The button will only appear if you have something in your cart already.

(This is extracted from https://github.com/Automattic/wp-calypso/pull/52890) 

Since this supersedes the `PopoverCart` component, that is removed by https://github.com/Automattic/wp-calypso/pull/57520 which should be merged only after this is available.

<img width="781" alt="Screen Shot 2021-12-14 at 3 03 07 PM" src="https://user-images.githubusercontent.com/2036909/146074547-a57bd615-5f51-40c0-a326-69a7b389a1a8.png">

<img width="325" alt="Screen Shot 2021-12-14 at 3 02 20 PM" src="https://user-images.githubusercontent.com/2036909/146074560-f62163cf-a5dc-4554-bf30-ef25b33a1006.png">


Note that this is only a calypso component, and is not available in the wp-admin versions of the Masterbar; support for that feature is planned but requires more work which will continue in https://github.com/Automattic/wp-calypso/pull/52890. In the meantime, this sets up the mini cart for reuse by putting it in its own npm package in the monorepo.

#### Testing instructions

- Clear your shopping cart.
- Verify that the cart button does not appear in the Masterbar.
- Add some products to your cart.
- Visit any page in calypso apart from checkout.
- Verify that the cart button appears in the Masterbar.
- Verify that the Masterbar cart button has a badge that denotes the number of items in the cart.
- Click the Masterbar cart button.
- Verify that you see the cart summary in a popup.
- Verify that clicking outside of the popup closes the summary.
- Verify that clicking the "remove product" link for each product removes the product.
- Verify that clicking the "Checkout" button redirects to checkout.
- Verify that the Masterbar cart button and the cart summary look good at mobile width as well as desktop width.
